PixelData documentation (#57)
NexTre-dev/retro-gadgets-docs: en/docs/modules/misc/VideoChip.md

### SetPixel(position `vec2`, color `color`)
Sets the pixel at the specified position to the specified **color**.

### SetPixelData(PixelData `PixelData`)

> ⚠️ **This feature is new, as of Retro Gadgets 0.1.5, and not fully tested yet.**  

Reviewer: Slight grammar change
```suggestion
⚠️ **This feature is new as of Retro Gadgets 0.1.5, and not fully tested yet.**  
```
We really should be testing things we add, but oh well.

Author: I did test all of the things I wrote, I just added that because it's new and undocumented so things might change in the future. Maybe it should be changed to something along the lines of: `This feature is new and undocumented as of 0.1.5, things might change in the future`

> Draws PixelData to the screen. Provided PixelData *needs* to be the same size as the VideoChip.

Reviewer: Slight grammar and formatting change
```suggestion
Draws PixelData to the screen. **Provided PixelData must be the same size as the VideoChip.**
```

Draws a render buffer (supposedly coming from **Webcam** component) at the desired position, width and height.

> ## `PixelData`

Reviewer: This is a new type! Might as well give it the proper treatment.
```suggestion
# `PixelData`
```

⚠️ **This feature is new, as of Retro Gadgets 0.1.5, and not fully tested yet.**  

> <!-- I don't know if this is the best description for it -->
> PixelData is a sort of buffer you can write to, and later display.
> It's a faster way to draw indivdual pixels to the screen compared to `SetPixel` on the VideoChip.

Reviewer: How's this for a description?
```suggestion
PixelData is a buffer you can draw to, similar to the VideoChip itself.  
As PixelData lives entirely in Lua, draw calls on it (such as `SetPixel`) are much faster the same call on the VideoChip. You can later draw the entire PixelData to a VideoChip at once.
```

> ### new(width `number`, height `number`, color `color`)

Reviewer: Obey header hierarchy (# -> ## -> ###, etc)
```suggestion
## Constructors

### new(width `number`, height `number`, color `color`)
```

> Constructor used to create PixelData. `width` and `height` represent the size of it, while `color` defines the color with which it is initialized.

Reviewer: No need to explain obvious parameters.
```suggestion
Creates a new instance of PixelData. `color` defines the color every pixel will be when initialized.
```

> ### After creating a PixelData you can use methods below to manipulate it.

Reviewer: Yeah that's what a category is.
```suggestion
## Methods
```

### Clear(color `color`)
Clears the PixelData to a specified color.

> ### SetPixel(x `number`, y `number`, color `color`) --> `color`

Reviewer: The current style omits any sort of return type symbology, based of the style used by the in-game multitool.
```suggestion
### SetPixel(x `number`, y `number`, color `color`) `color`
```
Think we should have `-->`? Open an issue to discuss with the community!

Sets the specified pixel to the color provided.

> ### GetPixel(x `number`, y `number`) --> `color`

Reviewer: ```suggestion
### GetPixel(x `number`, y `number`) `color`
```

Returns the color of a pixel on the `x` and `y` axis specified.

> 

Reviewer: ```suggestion

## Properties

```
Also: this should be above methods(?)

### Height - `number` **[Read only]**
<!-- Making it readonly because when you try to write to the variable, it demands `userdata` instead of number -->

> Height of the PixelData

Reviewer: ```suggestion
Height of the PixelData.
```

### Width - `number` **[Read only]**
<!-- Same as above-->

> Width of the PixelData

Reviewer: ```suggestion
Width of the PixelData.
```

## Example usage
<img src="./../../../assets/docs/VideoChip/PixelData.gif" width="250" align="right">

```lua
local vc = gdt.VideoChip0
local pixelData = PixelData.new(128, 128, color.clear) -- Must be the same size as the VideoChip

function update()
	for x = 1, pixelData.Width do
		for y = 1, pixelData.Height do
			pixelData:SetPixel(x, y, Color(math.random() * 256, math.random() * 256, math.random() * 256))
		end
	end
	vc:SetPixelData(pixelData)
end

> ```

Reviewer: # `MY EYES`

...Perhaps use a more pleasant example?
